File: CPP/Program/Vehicle.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;



class Vehicle {
private:
    string plat_nomor;
    string merk;
    int tahun_produksi;
    string warna;
    
public:
    // Constructors
    Vehicle() {

    }

    Vehicle(string plat_nomor, string merk, int tahun_produksi, string warna) {
        this->plat_nomor = plat_nomor;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        this->warna = warna;
    }

    // Getter methods
    string getPlatNomor() { 
        return plat_nomor; 
    }

    string getMerk() { 
        return merk; 
    }

    int getTahunProduksi() { 
        return tahun_produksi; 
    }

    string getWarna() { 
        return warna; 
    }

    
    // Setter methods
    void setPlatNomor(string plat_nomor) { 
        this->plat_nomor = plat_nomor; 
    }

    void setMerk(string merk) { 
        this->merk = merk;
    }

    void setTahunProduksi(int tahun_produksi) { 
        this->tahun_produksi = tahun_produksi; 
    }

    void setWarna(string warna) { 
        this->warna = warna; 
    }

    virtual void DisplayDetails() const {
        cout << "Plat Nomor: " << plat_nomor << endl;
        cout << "Merk: " << merk << endl;
        cout << "Tahun Produksi: " << tahun_produksi << endl;
        cout << "Warna: " << warna << endl;
    }

    // Destructor
    ~Vehicle() {

    }
};

File: Python/Program/Vehicle.py
class Vehicle:
    __platNomor = ""
    __merk = ""
    __tahunProduksi = ""
    __warna = ""

    def __init__(self, platNomor = "", merk = "", tahunProduksi = "", warna = ""):
        self. __platNomor = platNomor
        self. __merk = merk
        self.__tahunProduksi = tahunProduksi
        self.__warna = warna

    def getPlatNomor(self):
        return self.__platNomor
    
    def setPlatNomor(self, platNomor):
        self.__platNomor = platNomor

    def getMerk(self):
        return self.__merk

    def setMerk(self, merk):
        self.__merk = merk

    def getTahunProduksi(self):
        return self.__tahunProduksi
    
    def setTahunProduksi(self, tahunProduksi):
        self.__tahunProduksi = tahunProduksi

    def getWarna(self):
        return self.__warna
    
    def setWarna(self, warna):
        self.__warna = warna